src/components/external-events/ExternalEventsTable.svelte

```diff
@@ -0,0 +1,106 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import type { ValueGetterParams } from 'ag-grid-community';
+  import { plugins } from '../../stores/plugins';
+  import type { DataGridColumnDef } from '../../types/data-grid';
+  import type { ExternalEvent, ExternalEventId } from '../../types/external-event';
+  import type { ExternalSourceSlim } from '../../types/external-source';
+  import { getRowIdExternalEventWhole } from '../../utilities/externalEvents';
+  import { formatDate } from '../../utilities/time';
+  import DataGrid from '../ui/DataGrid/DataGrid.svelte';
+
+  export let selectedItemId: ExternalEventId | null;
+  export let items: ExternalEvent[];
+  export let filterExpression = '';
+
+  const baseColumnDefs: DataGridColumnDef[] = [
+    {
+      field: 'pkey',
+      filter: 'text',
+      headerName: 'Key',
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalEvent>) => {
+        if (params.data?.pkey) {
+          return params.data.pkey.key;
+        }
+      },
+    },
+    {
+      field: 'event_type_name',
+      filter: 'text',
+      headerName: 'Event Type',
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalEvent>) => {
+        if (params.data?.pkey) {
+          return params.data.pkey.event_type_name;
+        }
+      },
+    },
+    {
+      field: 'pkey',
+      filter: 'number',
+      headerName: 'Source Key',
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalEvent>) => {
+        if (params.data?.pkey) {
+          return params.data.pkey.source_key;
+        }
+      },
+    },
+    {
```

Comment:
I know that practically $plugins doesn't really change, but because it is a writable store, it is possible that it can change. This column would need to be added to `columnDefs` in a reactive statement to properly handle any potential updates.